common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.h

```diff
@@ -73,6 +73,31 @@ class sta_wlan_hal_nl80211 : public base_wlan_hal_nl80211, public sta_wlan_hal {
     }
 
 private:
+    // Client connection status
+    struct ConnectionStatus {
+        std::string bssid;
+        int freq;
+        std::string ssid;
+        int id;
+        std::string mode;
+        std::string pairwise_cipher;
+        std::string group_cipher;
+        std::string key_mgmt;
+        std::string wpa_state;
+        std::string address;
+        std::string uuid;
+    };
+
+    int add_network();
+    bool set_network(int network_id, const std::string &param, const std::string &value);
+    bool set_network_params(int network_id, const std::string &ssid, const std::string &bssid,
+                            WiFiSec sec, bool mem_only_psk, const std::string &pass,
+                            bool hidden_ssid, int freq = 0);
+    bool read_status(ConnectionStatus &connection_status);
+    void update_status(const ConnectionStatus &connection_status);
+    bool enable_network(int network_id);
+    bool is_connected(const std::string &wpa_state);
+
```

Comment:
please add Doxygen to all APIs

Reply:
I wish you'd been in the team when this code was originally written.